### PR TITLE
geant4: add v11.3.0

### PR DESCRIPTION
**Pull request by wdconinc**
This PR adds `geant4`, v11.3.0 ([release notes](https://geant4.web.cern.ch/download/release-notes/notes-v11.3.0.html)). This removes the preference for 11.2.2 (which was there to avoid picking the beta).

For now I have made the zeroth order assumption that there are no changes needed between 11.3.0.beta and 11.3.0 as released, but I'll go through that again later.

### CODE REVIEW

**Comment by wdconinc** [comment]
@drbenmorgan FYI, the release notes as linked from https://geant4.web.cern.ch/download/11.3.0.html are an empty page.

**Comment by drbenmorgan** [comment]
@wdconinc, will take a look at the empty page. If you can give me ten minutes, I have a build running with all the updated data libraries etc. I'll push and point you to my branch when this is confirmed to run and you're welcome to pull/incorporate that into this PR (to avoid duplication..)

**Comment by wdconinc** [comment]
If you're already have a branch, feel free to just close this PR as well :-) I have to go teach; back in an hour.

**Comment by drbenmorgan** [comment]
I think let's keep this PR as the main one as it's already cross-referenced elsewhere. My branch is here:

https://github.com/drbenmorgan/spack/tree/geant4-11.3

There's only 1 commit, and shouldn't conflict too badly with yours!

**Comment by drbenmorgan** [comment]
... and blank page for release notes should now be fixed, thanks for the report here!

**Comment by wdconinc** [comment]
> There's only 1 commit, and shouldn't conflict too badly with yours!

Cherry-picked and minor conflict resolved.

File: var/spack/repos/builtin/packages/g4channeling/package.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+    def setup_dependent_run_environment(self, env, dependent_spec):
+        install_path = join_path(self.prefix.share, "data", self.g4datasetname)
+        env.set("G4CHANNELINGDATA", install_path)
+
+    def url_for_version(self, version):
+        """Handle version string."""
+        return (
+            "http://geant4-data.web.cern.ch/geant4-data/datasets/G4CHANNELING.%s.tar.gz" % version
+        )
```
Comment (sethrj) [inline var/spack/repos/builtin/packages/g4channeling/package.py]: Now that spack requires python 3.6...
```suggestion
        return f"http://geant4-data.web.cern.ch/geant4-data/datasets/G4CHANNELING.{version}.tar.gz"
```

File: var/spack/repos/builtin/packages/g4channeling/package.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+    @property
+    def g4datasetname(self):
+        spec = self.spec
+        return "G4CHANNELING{0}".format(spec.version)
```
Comment (sethrj) [inline var/spack/repos/builtin/packages/g4channeling/package.py]:
```suggestion
        return f"G4CHANNELING{self.spec.version}"
```

**Review by sethrj — APPROVED** [review]
Thanks guys!

**Review by tldahlgren — COMMENTED** [review]
Confirmed homepage of new package and all affected version sha256s.

**Pull request merged**